blas/unit_test/Test_Blas1_rotmg.hpp

```diff
@@ -0,0 +1,245 @@
+      Kokkos::deep_copy(y1, 10.0);
+
+      ref_vals(0) = 45.0e-11 * gamma * gamma;
+      ref_vals(1) = 15.0e-3;
+      ref_vals(2) = 4.0e5 / (3 * gamma);
+      ref_vals(3) = 10.0;
+      ref_vals(4) = -1.0;
+      ref_vals(5) = 1.0 / gamma;
+      ref_vals(6) = -1.0e-4;
+      ref_vals(7) = 1.0e4 / (3 * gamma);
+      ref_vals(8) = 1.0;
+      break;
+    case 5:
+      Kokkos::deep_copy(d1, 4.0e10);
+      Kokkos::deep_copy(d2, 2.0e-2);
+      Kokkos::deep_copy(x1, 1.0e-5);
+      Kokkos::deep_copy(y1, 10.0);
+
+      ref_vals(0) = 4.0e10 / (1.5 * gamma * gamma);
+      ref_vals(1) = 2.0e-2 / 1.5;
+      ref_vals(2) = 6144.0e-5;
+      ref_vals(3) = 10.0;
+      ref_vals(4) = -1.0;
+      ref_vals(5) = 4096.0;
+      ref_vals(6) = -1.0e6;
+      ref_vals(7) = 5.0e-7 * gamma;
+      ref_vals(8) = 1.0;
+      break;
+    case 6:
+      Kokkos::deep_copy(d1, 2.0e-10);
+      Kokkos::deep_copy(d2, 4.0e-2);
+      Kokkos::deep_copy(x1, 1.0e5);
+      Kokkos::deep_copy(y1, 10.0);
+
+      ref_vals(0) = 4.0 / 150.0;
+      ref_vals(1) = (2.0e-10 / 1.5) * (gamma * gamma);
+      ref_vals(2) = 15.0;
+      ref_vals(3) = 10.0;
+      ref_vals(4) = -1.0;
+      ref_vals(5) = 5.0e-5;
+      ref_vals(6) = -1.0 / gamma;
+      ref_vals(7) = 1.0;
+      ref_vals(8) = 1.0e4 / gamma;
+      break;
+    case 7:
+      Kokkos::deep_copy(d1, 2.0e10);
+      Kokkos::deep_copy(d2, 4.0e-2);
+      Kokkos::deep_copy(x1, 1.0e-5);
+      Kokkos::deep_copy(y1, 10.0);
+
+      ref_vals(0) = 4.0 / 150.0;
+      ref_vals(1) = 2.0e10 / (1.5 * gamma * gamma);
+      ref_vals(2) = 15.0;
+      ref_vals(3) = 10.0;
+      ref_vals(4) = -1.0;
+      ref_vals(5) = 5.0e5;
+      ref_vals(6) = -4096.0;
+      ref_vals(7) = 1.0;
+      ref_vals(8) = 4096.0e-6;
+      break;
+    case 8:
+      Kokkos::deep_copy(d1, 4.0);
+      Kokkos::deep_copy(d2, -2.0);
+      Kokkos::deep_copy(x1, 8.0);
+      Kokkos::deep_copy(y1, 4.0);
+
+      ref_vals(0) = 32.0 / 7.0;
+      ref_vals(1) = -16.0 / 7.0;
+      ref_vals(2) = 7.0;
+      ref_vals(3) = 4.0;
+      ref_vals(4) = 0.0;
+      ref_vals(5) = 0.0;
+      ref_vals(6) = -0.5;
+      ref_vals(7) = -0.25;
+      ref_vals(8) = 0.0;
+      break;
+    default:
+      throw std::runtime_error("rotmg test: test case unrecognized!");
+      break;
+  }
+}
+}  // namespace Test
+
+template <class Scalar, class ExecutionSpace>
+int test_rotmg() {
+  Kokkos::View<Scalar, Kokkos::Device<ExecutionSpace,
+                                      typename ExecutionSpace::memory_space>>
+      d1("d1"), d2("d2"), x1("x1"), y1("y1");
+  Kokkos::View<Scalar[5], Kokkos::Device<ExecutionSpace,
+                                         typename ExecutionSpace::memory_space>>
+      param("param");
+  Kokkos::View<Scalar[9], Kokkos::DefaultHostExecutionSpace> ref_vals(
+      "reference values");
+
+  constexpr int num_test_cases = 9;
+  for (int test_case = 0; test_case < num_test_cases; ++test_case) {
+#if defined(KOKKOSKERNELS_ENABLE_TPL_ARMPL)
```

Comment:
Can take this if defined branch out if ARMPL is never used

Reply:
Sure, I removed that `#ifdef` since it is no longer required.